### PR TITLE
Fast import of initial YML data files

### PR DESCRIPTION
**Pull request by jacobchristensen79**

### CODE REVIEW

**Comment by hairmare** [comment]
What does this do that https://github.com/libgraviton/import-export can't do? The tool already has a facility to load data directly into MongoDB. That importing the yaml files is slow is due to the fact that it does so via HTTP. This is an important feature because it ensures that we only add data through the yaml files that can be changed after it is deployed on production...

File: src/Graviton/CoreBundle/Command/DirectImportCommand.php [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+        // Do this by rest
+        $restObjects = [];
+
+        /** @var SplFileInfo $file */
+        foreach ($finder as $file) {
+            $fileName = str_replace($location, '', $file->getPathname());
+
+            $progress->advance();
+
+            $contents = explode('---', $file->getContents());
```
Comment (hairmare) [inline src/Graviton/CoreBundle/Command/DirectImportCommand.php]: There is `webuni/front-matter` that does frontmatter like a baus.

File: src/Graviton/CoreBundle/Command/DirectImportCommand.php [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+            if (!$contents || count($contents)!==3) {
+                $errors[$fileName] = 'Content of file is incorrect, could not parse it with --- ';
+                continue;
+            }
+
+            $target = trim(str_replace('target:', '', $contents[1]));
```
Comment (hairmare) [inline src/Graviton/CoreBundle/Command/DirectImportCommand.php]: We might introduce a `file:` attribute in frontmatter for loading to `/file` soonish so this needs to become more general.

**Comment by narcoticfresh** [comment]
@jacobchristensen79 @hairmare 
was there an answer why we do this while we have `graviton-import-export`?
as there is no PR description it's hard to see what the purpose is(?)

**Comment by hairmare** [comment]
@narcoticfresh There is an error when loading initialdata with graviton-import-export in vagrant boxes. I'm currently trying to figure out why the built-in server hangs up after a while and then only responds with 500.

**Comment by narcoticfresh** [comment]
@hairmare  did the vagrant users try to load them synchronously (`-s`)?

**Comment by hairmare** [comment]
@narcoticfresh yes, nevertheless the built in server just breaks down after loading about half of the files ;(

File: src/Graviton/CoreBundle/Command/DirectImportCommand.php (hunk 2 above)
Comment (jacobchristensen79) [inline src/Graviton/CoreBundle/Command/DirectImportCommand.php]: Agree, but I think File should not be in target but more generic to the item to be inserted ( As we also have items and file may be also files...)

File: src/Graviton/RestBundle/Validator/Form.php [hunk 3]
```diff
@@ -101,6 +103,48 @@ public function checkForm(
...
+                if (!$child->isValid()) {
+                    $errors[$child->getName()] = $this->getErrorMessages($child);
+                }
+            }
+        }
+
+        return $errors;
+    }
+
+    /**
```
Comment (jacobchristensen79) [inline src/Graviton/RestBundle/Validator/Form.php]: Now we get all form errors, not only something went wrong but what.

**Comment by jacobchristensen79** [comment]
I have tested this importing the new initial data. It's working but we have to review the data files for incorrect params.

**Pull request closed without merging**